Refactoring of `TrialState` and `FrozenTrial`. (#1101)
optuna/optuna: docs/source/reference/structs.rst

.. autoclass:: TrialState
    :members:

    .. deprecated:: 1.4.0

>             The use of this class is deprecated and it is recommended that you use :class:`~optuna.trial.TrialState` instead.

Reviewer: ```suggestion
            This class is deprecated. Please use :class:`~optuna.trial.TrialState` instead.
```
